core/templates/pages/about-page/about-page.component.ts

```diff
@@ -19,100 +19,79 @@
-  listOfNamesToThank = [
-    'Alex Kauffmann', 'Allison Barros',
-    'Amy Latten', 'Brett Barros',
-    'Crystal Kwok', 'Daniel Hernandez',
-    'Divya Siddarth', 'Ilwon Yoon',
-    'Jennifer Chen', 'John Cox',
-    'John Orr', 'Katie Berlent',
-    'Michael Wawszczak', 'Mike Gainer',
-    'Neil Fraser', 'Noah Falstein',
-    'Nupur Jain', 'Peter Norvig',
-    'Philip Guo', 'Piotr Mitros',
-    'Rachel Chen', 'Rahim Nathwani',
-    'Robyn Choo', 'Tricia Ngoon',
-    'Vikrant Nanda', 'Vinamrata Singal',
-    'Yarin Feigenbaum'];
-  // Define constant for each tab on the page.
-  TAB_ID_ABOUT = 'about';
-  TAB_ID_FOUNDATION = 'foundation';
-  TAB_ID_CREDITS = 'credits';
-  ALLOWED_TABS = [
-    this.TAB_ID_ABOUT, this.TAB_ID_FOUNDATION, this.TAB_ID_CREDITS];
+  classroomUrlFragment: string;
+  classroomUrl :string;
+  loginUrl: string;
+  userIsLoggedIn: boolean;
   constructor(
     private urlInterpolationService: UrlInterpolationService,
-    private windowRef: WindowRef) {
+    private windowRef: WindowRef,
+    private siteAnalyticsService: SiteAnalyticsService,
+    private userService: UserService,
+    private loaderService: LoaderService) {
   }
 
-  getCredits(startLetter: string): string[] {
-    const results = AboutPageConstants.CREDITS_CONSTANTS.filter(
-      (credit) => credit.startsWith(startLetter)).sort();
-    return results;
+  getStaticImageUrl(imagePath: string): string {
+    return this.urlInterpolationService.getStaticImageUrl(imagePath);
   }
 
-  onTabClick(tabName: string): Window {
-    this.windowRef.nativeWindow.location.hash = '#' + tabName;
-    this.activeTabName = tabName;
-    return this.windowRef.nativeWindow;
+  onClickVisitClassroomButton(): void {
+    this.siteAnalyticsService.registerClickVisitClassroomButtonEvent();
+    this.windowRef.nativeWindow.location.href = this.classroomUrl;
   }
 
-  getStaticImageUrl(imagePath: string): string {
-    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  onClickBrowseLibraryButton(): void {
+    this.siteAnalyticsService.
+      registerClickBrowseLibraryButtonEvent();
+    this.windowRef.nativeWindow.location.href = ('/community-library');
   }
 
-  ngOnInit(): void {
-    this.activeTabName = this.TAB_ID_ABOUT;
-    this.allCredits = [];
-    const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
-    for (const letter of letters) {
-      const names = this.getCredits(letter);
-      if (names.length > 0) {
-        this.allCredits.push({letter, names});
-      }
-    }
-    const hash = this.windowRef.nativeWindow.location.hash.slice(1);
-    if (hash === 'license') {
-      this.activeTabName = this.TAB_ID_FOUNDATION;
-    } else if (this.ALLOWED_TABS.includes(hash)) {
-      this.activeTabName = hash;
+  onClickCreateLessonButton(): void {
+    this.siteAnalyticsService.registerCreateLessonButtonEvent();
+    if (this.userIsLoggedIn === null) {
+      this.windowRef.nativeWindow.location.href = this.loginUrl;
+    } else {
```

Comment:
What if `this.userIsLoggedIn` is false? will thsi function redirect to `'/creator-dashboard?mode=create`?

Reply:
yes, If user is not logged in it will first open the login window for user to login and once the user is logged in it will redirect to '/creator-dashboard?mode=create